### PR TITLE
feat:[HL-83] List All Categories, Area and Ingredients API

### PR DESCRIPTION
**Pull request by mhmdatallaa**
#### Proposed Changes

As title

#### Related Issues

N/A

#### Additional Information

N/A

#### Screenshot

N/A

#### Checklist

- [x] I have tested these changes locally.
- [x] I have added appropriate documentation or updated existing documentation.
- [x] I have added appropriate test coverage or updated existing test coverage.
- [ ] I have updated the changelog (if applicable).
- [x] I have followed the project's code style and formatting guidelines.
- [x] I have reviewed and adhered to the project's contributing guidelines.

### CODE REVIEW

**Review by esraakhaled — APPROVED** [review]

**Review by abdallah-7698 — APPROVED** [review]

File: Networking/Sources/Networking/Requests/ACIListRequests/CategoriesListRequest.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+// MARK: - CategoriesListResponse
+public struct CategoriesListResponse: Decodable {
+    let meals: [CategoryMeal]
+}
+
+// MARK: - Meal
+struct CategoryMeal: Decodable {
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/ACIListRequests/CategoriesListRequest.swift]: What is a category meal? It doesn't reflect what it means. 

A good SWE should write an API that is readable from his SELF Future and for the team.

File: Networking/Sources/Networking/Requests/ACIListRequests/AreaListRequest.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    enum CodingKeys: String, CodingKey {
+        case area = "strArea"
+    }
+}
+
+// MARK: - AreaListRequest
+public struct AreaListRequest: RequestType {
+    public typealias ResponseType = AreaListResponse
+
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/ACIListRequests/AreaListRequest.swift]:
```suggestion
```
Type inference would save us in this case.

File: Healthy/Classes/Entities/Category.swift [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+// MARK: - Category
+public struct CategoryList {
+    let meals: [Category]
+}
+
+// MARK: - Meal
+struct Category {
+    let category: String
+}
```
Comment (ahmdmhasn) [inline Healthy/Classes/Entities/Category.swift]:
```suggestion
// MARK: - Category
struct Category {
    let title: String
}
```

File: Healthy/Classes/Entities/Ingrediant.swift [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    let meals: [Meal]
+}
+
+// MARK: - Meal
+public struct Meal {
+    let id: String
+    let ingrediant: String
+    let description: String?
+    let type: String?
+}
```
Comment (ahmdmhasn) [inline Healthy/Classes/Entities/Ingrediant.swift]:
```suggestion
// MARK: - Meal
struct Meal {
    let id: String
    let ingrediant: String
    let description: String?
    let type: String?
}
```

File: Healthy/Classes/Entities/Area.swift [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+// MARK: - Area
+public struct AreaList {
+    let meals: [Area]
+}
+
+// MARK: - Meal
+struct Area: Codable {
+    let area: String
+}
```
Comment (ahmdmhasn) [inline Healthy/Classes/Entities/Area.swift]:
```suggestion
// MARK: - Meal
struct Area {
    let title: String
}
```

File: Networking/Sources/Networking/Requests/ACIListRequests/AreaListRequest.swift [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+// MARK: - AreaListResponse
+public struct AreaListResponse: Decodable {
+    let meals: [AreaMeal]
+}
+
+// MARK: - Meal
+struct AreaMeal: Decodable {
+    let area: String
+
+    enum CodingKeys: String, CodingKey {
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/ACIListRequests/AreaListRequest.swift]:
```suggestion
    private enum CodingKeys: String, CodingKey {
```

File: Networking/Sources/Networking/Requests/ACIListRequests/AreaListRequest.swift [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+// MARK: - AreaListResponse
+public struct AreaListResponse: Decodable {
+    let meals: [AreaMeal]
+}
+
+// MARK: - Meal
+struct AreaMeal: Decodable {
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/ACIListRequests/AreaListRequest.swift]:
```suggestion
// MARK: - MealArea
struct MealArea: Decodable {
```

File: Networking/Sources/Networking/Requests/ACIListRequests/CategoriesListRequest.swift (hunk 1 above)
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/ACIListRequests/CategoriesListRequest.swift]:
```suggestion
// MARK: - MealCategory
struct MealCategory: Decodable {
```

**Review by ahmdmhasn — APPROVED** [review]

File: Healthy/Classes/Entities/Area.swift [hunk 8]
```diff
@@ -0,0 +1,4 @@
+// MARK: - Meal
+struct Area: Codable {
```
Comment (ahmdmhasn) [inline Healthy/Classes/Entities/Area.swift]:
```suggestion
struct Area {
```

This entity is not meant to be encoded or decoded.

**Review by ahmdmhasn — APPROVED** [review]

**Pull request merged**